keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensorTest.java

```diff
@@ -839,9 +836,124 @@ public void youCanFixACustomValidationIssueByReplacingTheValue() {
     @Test
     public void comparesDoubleTensorWithScalar() {
         DoubleTensor value = DoubleTensor.create(1., 2., 3.);
-        DoubleTensor differentValue = DoubleTensor.create(1.);
+        DoubleTensor differentValue = DoubleTensor.scalar(1.);
         BooleanTensor result = value.elementwiseEquals(differentValue);
         assertThat(result, hasValue(true, false, false));
     }
 
+    @Test
+    public void comparesDoubleTensorWithLength1Vector() {
+        DoubleTensor value = DoubleTensor.create(new double[]{1.}, 1);
```

Comment:
`DoubleTensor.create(1.)`

Reply:
I wanted to be explicit about the length 1 by explicitly specifying the shape.